tools/Google.Cloud.Tools.ReleaseManager/History/HistoryFile.cs

```diff
@@ -132,18 +134,44 @@ internal Section(Release release)
                 }
                 else
                 {
-                    foreach (var commit in release.Commits)
-                    {
-                        Lines.AddRange(commit.GetHistoryLines());
-                    }
+                    Lines.AddRange(GetNotesFromCommits(release.Commits));
                     // No "interesting" commits? That usually means it's just a dependency update.
                     if (Lines.Count == 2)
                     {
                         Lines.Add("No API surface changes; just dependency updates.");
+                        Lines.Add("");
                     }
-                    Lines.Add("");
                 }
             }
+
+            internal IEnumerable<string> GetNotesFromCommits(IReadOnlyList<GitCommit> commits) =>
+                commits
+                    .SelectMany(c => c.GetReleaseNoteElements())
+                    .ToLookup(e => e.Type)
+                    .Where(g => g.Key != ReleaseNoteElementType.Chore)
+                    .OrderBy(g => g.Key)
+                    .Select(GetNotesFromElement)
+                    .SelectMany(line => line);
+
+            internal IEnumerable<string> GetNotesFromElement(IGrouping<ReleaseNoteElementType, ReleaseNoteElement> elements)
+            {
+                string description = elements.Key switch
+                {
+                    ReleaseNoteElementType.Fix => "Bug fixes",
+                    ReleaseNoteElementType.Feature => "New features",
+                    ReleaseNoteElementType.Chore => "Chores",
+                    ReleaseNoteElementType.Docs => "Documentation improvements",
+                    ReleaseNoteElementType.Unknown => FixmeBlockingRelease,
+                    _ => throw new InvalidOperationException("Unknown element type")
```

Comment:
This should signal a bug in the tool, right? Unknown elements sould really be written out on the Fixme section?

Reply:
Yes, it's a bug. This is where "we don't know what the element type is" rather than "we know that the element type is 'unknown', because we parsed it to that". Really, really shouldn't happen :)